### PR TITLE
Constrain dimension of equalityConstraint

### LINKED ISSUE #2886: Clarify known size in equalityConstraint
In the definition of the `equalityConstraint` function in 9.4.1 [Overconstrained Equation Operators for Connection Graphs](https://specification.modelica.org/maint/3.5/connectors-and-connections.html#overconstrained-equation-operators-for-connection-graphs) it says:
> The residue output of the equalityConstraint function shall have known size, say constant .

What is "known size" in this case? In the discussion of #2882 it was mentioned [it would normally be replaced by a number](https://github.com/modelica/ModelicaSpecification/pull/2882#discussion_r586229575). I thought "known size" means it can be evaluated at the translation time. I believe it needs to be clarified.

### PR DESCRIPTION
**Pull request by henrikt-ma**
Fixes #2886.

According to phone meeting poll: https://github.com/modelica/ModelicaSpecification/issues/2886#issuecomment-800323151

### CODE REVIEW

**Review by HansOlsson — APPROVED** [review]
ok

File: chapters/connectors.tex [hunk 1]
```diff
@@ -813,7 +813,9 @@ \subsection{Overconstrained Equation Operators for Connection Graphs}\label{over
   end equalityConstraint;
 end Record;
 \end{lstlisting}
-The \lstinline!residue! output of the \lstinline!equalityConstraint! function shall have known size, say constant $n$.  The function shall express the equality between the two type instances \lstinline!T1! and \lstinline!T2! or the record instances \lstinline!R1! and \lstinline!R2!, respectively, with a non-redundant number $n \ge 0$ of equations.  The residues of these equations are returned in vector \lstinline!residue! of size $n$.  The set of $n$ non-redundant equations stating that \lstinline!R1 = R2! is given by the equation (\lstinline!0! represents a vector of zeros of appropriate size):
+The array dimension $n$ of \lstinline!residue! shall be a constant \lstinline!Integer! expression that can be evaluated during translation, with $n > 0$.
```
Comment (eshmoylova) [inline chapters/connectors.tex:816]: Sorry, I am too late with my review.  I see the restriction of 'n > 0' was added. There is at least one Modelica library [that sets it to 0](https://github.com/lbl-srg/modelica-buildings/blob/bddfd49d63a9c056073551b1267591be680e4852/Buildings/Electrical/PhaseSystems/PartialPhaseSystem.mo#L30).  @mwetter

Comment (henrikt-ma) [reply chapters/connectors.tex:816]: Excellent observation.  Will fix.

Comment (henrikt-ma) [reply chapters/connectors.tex:816]: Having opened #2909, I'll copy the comments above to that conversation and mark this conversation as resolved.

**Pull request merged**